### PR TITLE
PR: Enable running renamed and IPython files again

### LINKED ISSUE #20701: Can not run renamed file in master
To reproduce:
1 - create unsaved new file
Screenshot 2023-03-18 at 07 54 01

2 - run it with the run button. See the expected result

3 - save it on the desktop under "test.py". The name of the file changed but not the displayed path
Screenshot 2023-03-18 at 07 54 19

4 - run it again, Spyder tries to run the old filename, which fails
```
runfile('/Users/quentin/untitled4.py', wdir='/Users/quentin')
This command failed to be executed because an error occurred while trying to get the file code from Spyder's editor. The error was:


An exception has occurred, use %tb to see the full traceback.

FileNotFoundError: [Errno 2] No such file or directory: '/Users/quentin/untitled4.py'
```

5 - focus on the console and back on the editor. The run button is now disabled but the path is correct.
Screenshot 2023-03-18 at 07 54 38

6 - close and reopen the tab, everything works fine
Screenshot 2023-03-18 at 07 59 58
7 - rename in the file plugin `test.py` to `test2.py`. The run button is disabled again
Screenshot 2023-03-18 at 08 00 14

### LINKED ISSUE #20715: Can not run IPython files in master
When opening a .ipy file with the master, the run button is grayed out:
Screenshot 2023-03-22 at 07 16 35

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes
 - Removes `first_execution` so there is no need to confirm every time a new file is run.
 - adds a test for running files with quotes in their names (see https://github.com/spyder-ide/spyder/pull/20633)
 - Support running ".ipy" files as well as ".py" files
 - Removes `sig_switch_run_configuration_focus` whih was unused

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #20701
Fixes #20715

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/app/tests/test_mainwindow.py [hunk 1]
```diff
@@ -6331,5 +6331,94 @@ def test_runfile_namespace(main_window, qtbot, tmpdir):
     assert "test_globals True" in control.toPlainText()
 
 
+@pytest.mark.skipif(
+    os.name == 'nt',
+    reason="No quotes in windows filepath"
```
Comment (ccordoba12) [inline spyder/app/tests/test_mainwindow.py:6336]:
```suggestion
    reason="No quotes on Windows file paths"
```

File: spyder/app/tests/test_mainwindow.py [hunk 2]
```diff
@@ -6331,5 +6331,94 @@ def test_runfile_namespace(main_window, qtbot, tmpdir):
...
+
+    with qtbot.waitSignal(shell.executed):
+        qtbot.mouseClick(main_window.run_cell_button, Qt.LeftButton)
+
+    assert "802" in control.toPlainText()
+    assert "error" not in control.toPlainText()
+
+    # Make sure this works with ipy and renamed files too
+
+    # Rename the file to ipython and send the signal
```
Comment (ccordoba12) [inline spyder/app/tests/test_mainwindow.py:6374]:
```suggestion
    # Rename the file to IPython and emit the signal for that
```

File: spyder/app/tests/test_mainwindow.py [hunk 3]
```diff
@@ -6331,5 +6331,94 @@ def test_runfile_namespace(main_window, qtbot, tmpdir):
...
+    code_editor.set_text("print(21 + 780)")
+
+    with qtbot.waitSignal(shell.executed):
+        qtbot.mouseClick(main_window.run_button, Qt.LeftButton)
+
+    assert "801" in control.toPlainText()
+    assert "error" not in control.toPlainText()
+    assert "\\.ipy" in control.toPlainText()
+    
+    # Create an untiliteled file
```
Comment (ccordoba12) [inline spyder/app/tests/test_mainwindow.py:6388]:
```suggestion
    # Create an untitled file
```

File: spyder/app/tests/test_mainwindow.py [hunk 4]
```diff
@@ -6331,5 +6331,94 @@ def test_runfile_namespace(main_window, qtbot, tmpdir):
...
+    code_editor.set_text("print(20 + 780)")
+    
+    with qtbot.waitSignal(shell.executed):
+        qtbot.mouseClick(main_window.run_cell_button, Qt.LeftButton)
+
+    assert "800" in control.toPlainText()
+    assert "error" not in control.toPlainText()
+    assert "untitled" in control.toPlainText()
+    
+    # save in a new folder
```
Comment (ccordoba12) [inline spyder/app/tests/test_mainwindow.py:6403]:
```suggestion
    # Save file in a new folder
```

File: spyder/app/tests/test_mainwindow.py [hunk 5]
```diff
@@ -6331,5 +6331,94 @@ def test_runfile_namespace(main_window, qtbot, tmpdir):
...
+    code_editor.set_text("print(19 + 780)")
+    
+    with tempfile.TemporaryDirectory() as td:
+    
+        editorstack = main_window.editor.get_current_editorstack()
+        editorstack.select_savename = lambda fn: os.path.join(td, "fn.ipy")
+        main_window.editor.save()
+        
+
```
Comment (ccordoba12) [inline spyder/app/tests/test_mainwindow.py:6412]:
```suggestion

```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @impact27 for your work on this!

File: spyder/plugins/pylint/plugin.py [hunk 6]
```diff
@@ -86,15 +86,15 @@ def on_initialize(self):
         # Run configuration
         self.executor_configuration = [
             {
-                'input_extension': 'py',
+                'input_extension': ['py', 'ipy'],
```
[the remark this answers is not in the document]
Comment (impact27) [reply spyder/plugins/pylint/plugin.py:89]: I am not sure, it should be, as ipy files are python files + magics?

File: spyder/app/tests/test_mainwindow.py (hunk 1 above)
Comment (impact27) [reply spyder/app/tests/test_mainwindow.py]: I don't have a windows machine handy so maybe a test could be added for this at a later point?

File: spyder/plugins/pylint/plugin.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ccordoba12) [reply spyder/plugins/pylint/plugin.py]: I tested locally and Pylint can't handle IPython files.

**Comment by ccordoba12** [comment]
@impact27, I disabled running IPython files in the Code Analysis and External Console plugins because those plugins can't handle them.

I also pushed a fix for a bug I found while testing this locally: if you renamed a Python file to an IPython one in the editor, then it was registered twice, which caused a bug.

**Review by ccordoba12 — APPROVED** [review]
Thanks @impact27!

**Pull request merged**